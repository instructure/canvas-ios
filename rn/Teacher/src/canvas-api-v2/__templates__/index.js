// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
