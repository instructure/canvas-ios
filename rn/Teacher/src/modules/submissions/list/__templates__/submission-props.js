//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
