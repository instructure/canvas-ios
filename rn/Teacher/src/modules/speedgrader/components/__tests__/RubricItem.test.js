//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

import { shallow } from 'enzyme'
import React from 'react'
import { Alert, AccessibilityInfo, ActionSheetIOS } from 'react-native'
import RubricItem from '../RubricItem'
import * as templates from '../../../../__templates__'

jest.mock('react-native/Libraries/Alert/Alert', () => ({
  prompt: jest.fn(),
}))
jest.mock('react-native/Libraries/ActionSheetIOS/ActionSheetIOS', () => ({
  showActionSheetWithOptions: jest.fn(),
}))

let defaultProps = {
  rubricItem: templates.rubric(),
  grade: { comments: '' },
  showDescription: jest.fn(),
  changeRating: jest.fn(),
  openCommentKeyboard: jest.fn(),
  deleteComment: jest.fn(),
  freeFormCriterionComments: false,
}

describe('RubricItem', () => {
  beforeEach(() => jest.clearAllMocks())

  it('renders', () => {
    let tree = shallow(<RubricItem {...defaultProps} />)
    expect(tree.find('Text').first().prop('children')).toBe(defaultProps.rubricItem.description)
  })

  it('shows the text for not impacting a score if `ignore_for_scoring` is true', () => {
    let props = {
      ...defaultProps,
      rubricItem: templates.rubric({ ignore_for_scoring: true }),
    }
    let tree = shallow(<RubricItem {...props} />)
    expect(tree.find(`[testID='rubric-item.${props.rubricItem.id}-no-score']`).length).toEqual(1)
  })

  it('renders with free form', () => {
    let props = {
      ...defaultProps,
      freeFormCriterionComments: true,
    }
    let tree = shallow(<RubricItem {...props} />)
    let button = tree.find(`[testID='rubric-item.add-comment-${props.rubricItem.id}']`)
    expect(button.prop('children')).toBe('Add Comment')
  })

  it('renders an already selected grade', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 10,
        rating_id: '3',
        comments: '',
      },
    }
    let tree = shallow(<RubricItem {...props} />)
    expect(tree.find(`[testID='rubric-item.points-3']`).prop('on')).toBe(true)
    expect(tree.find(`[testID='rubric-item.points-3']`).prop('children')).toBe('10')
  })

  it('renders a custom grade on a selected rating', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 8,
        rating_id: '3',
        comments: '',
      },
    }
    let tree = shallow(<RubricItem {...props} />)
    expect(tree.find(`[testID='rubric-item.points-3']`).prop('on')).toBe(true)
    expect(tree.find(`[testID='rubric-item.points-3']`).prop('children')).toBe('8')
  })

  it('renders the selected custom grade', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 12,
        comments: '',
      },
    }
    let tree = shallow(<RubricItem {...props} />)
    let button = tree.find(`[testID='rubric-item.customize-grade-${props.rubricItem.id}']`)
    expect(button.prop('children')).toBe('12')
  })

  it('calls showDescription when the button is pressed', () => {
    let tree = shallow(<RubricItem {...defaultProps} />)
    tree.find(`[testID='rubric-item.description']`).simulate('Press')
    expect(defaultProps.showDescription).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('changes the currently selected value when a circle is pressed', () => {
    let tree = shallow(<RubricItem {...defaultProps} />)
    tree.find(`[testID='rubric-item.points-2']`).simulate('Press')
    expect(defaultProps.changeRating).toHaveBeenCalled()
  })

  it('removes a selection when a selected circle is pressed', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 10,
        rating_id: '3',
        comments: '',
      },
    }
    let tree = shallow(<RubricItem {...props} />)
    expect(tree.find(`[testID='rubric-item.points-3']`).prop('on')).toBe(true)
    tree.find(`[testID='rubric-item.points-3']`).simulate('Press')
    expect(tree.find(`[testID='rubric-item.points-3']`).prop('on')).toBe(false)
  })

  it('gets the value from prompting for a custom value', () => {
    let tree = shallow(<RubricItem {...defaultProps} />)
    tree.find(`[testID='rubric-item.customize-grade-${defaultProps.rubricItem.id}']`)
      .simulate('Press')

    expect(Alert.prompt).toHaveBeenCalled()
    expect(Alert.prompt.mock.calls[0][5]).toEqual('decimal-pad')
    Alert.prompt.mock.calls[0][2][1].onPress('12')
    expect(defaultProps.changeRating).toHaveBeenCalledWith(defaultProps.rubricItem.id, 12, undefined)
    expect(AccessibilityInfo.setAccessibilityFocus).toHaveBeenCalled()
  })

  it('refocuses the customize button on cancel of the prompt', () => {
    let tree = shallow(<RubricItem {...defaultProps} />)
    tree.find(`[testID='rubric-item.customize-grade-${defaultProps.rubricItem.id}']`)
      .simulate('Press')

    expect(Alert.prompt).toHaveBeenCalled()
    Alert.prompt.mock.calls[0][2][0].onPress()
    expect(AccessibilityInfo.setAccessibilityFocus).toHaveBeenCalled()
  })

  it('will remove the selected value if it is already set', () => {
    let props = {
      ...defaultProps,
      grade: {
        points: 1234,
        comments: '',
      },
    }
    let tree = shallow(<RubricItem {...props} />)
    tree.find(`[testID='rubric-item.customize-grade-${props.rubricItem.id}']`)
      .simulate('Press')
    expect(tree.state('selectedPoints')).toBe(null)
  })

  it('will call openCommentKeyboard when the add comment button is pressed', () => {
    let props = {
      ...defaultProps,
      freeFormCriterionComments: true,
    }
    let tree = shallow(<RubricItem {...props} />)
    tree.find(`[testID='rubric-item.add-comment-${props.rubricItem.id}']`)
      .simulate('Press')
    expect(defaultProps.openCommentKeyboard).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('will not show the add comment button and will show the comment if there is one', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }
    let tree = shallow(<RubricItem {...props} />)
    let button = tree.find(`[testID='rubric-item.add-comment-${props.rubricItem.id}']`)
    expect(button.exists()).toBe(false)
  })

  it('will open an action sheet when a rubric comment is pressed', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }
    let tree = shallow(<RubricItem {...props} />)
    tree.find(`[testID='rubric-item.edit-comment-${props.rubricItem.id}']`)
      .simulate('Press')
    expect(ActionSheetIOS.showActionSheetWithOptions).toHaveBeenCalled()
  })

  it('will do nothing when cancel is pressed in the edit action sheet', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }
    let tree = shallow(<RubricItem {...props} />)
    tree.find(`[testID='rubric-item.edit-comment-${props.rubricItem.id}']`)
      .simulate('Press')

    ActionSheetIOS.showActionSheetWithOptions.mock.calls[0][1](2)
    expect(defaultProps.deleteComment).not.toHaveBeenCalled()
    expect(defaultProps.openCommentKeyboard).not.toHaveBeenCalled()
  })

  it('will call deleteComment when the delete option is pressed in the edit action sheet', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }
    let tree = shallow(<RubricItem {...props} />)
    tree.find(`[testID='rubric-item.edit-comment-${props.rubricItem.id}']`)
      .simulate('Press')

    ActionSheetIOS.showActionSheetWithOptions.mock.calls[0][1](1)
    expect(defaultProps.deleteComment).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('will call openCommentKeyboard when the edit option is pressed in the edit action sheet', () => {
    let props = {
      ...defaultProps,
      grade: { comments: 'A comment' },
    }
    let tree = shallow(<RubricItem {...props} />)
    tree.find(`[testID='rubric-item.edit-comment-${props.rubricItem.id}']`)
      .simulate('Press')

    ActionSheetIOS.showActionSheetWithOptions.mock.calls[0][1](0)
    expect(defaultProps.openCommentKeyboard).toHaveBeenCalledWith(defaultProps.rubricItem.id)
  })

  it('calls show tooltip with the rating description', () => {
    const showToolTip = jest.fn()
    let tree = shallow(<RubricItem {...defaultProps} showToolTip={showToolTip} />)
    tree.find(`[testID='rubric-item.points-${defaultProps.rubricItem.ratings[0].id}']`)
      .simulate('LongPress', defaultProps.rubricItem.ratings[0].id, { x: 8, y: 9, width: 10, height: 44 })

    expect(showToolTip).toHaveBeenCalledWith(
      { x: 13, y: 9 },
      defaultProps.rubricItem.ratings[0].description
    )
  })

  it('calls dismissToolTip onPressOut', () => {
    const dismissToolTip = jest.fn()
    let tree = shallow(<RubricItem {...defaultProps} dismissToolTip={dismissToolTip} />)
    tree.find(`[testID='rubric-item.points-${defaultProps.rubricItem.ratings[0].id}']`)
      .simulate('PressOut')
    expect(dismissToolTip).toHaveBeenCalled()
  })
})
