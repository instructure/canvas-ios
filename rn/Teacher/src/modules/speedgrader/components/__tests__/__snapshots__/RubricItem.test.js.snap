// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubricItem changes the currently selected value when a circle is pressed 1`] = `
<View
  style={
    Object {
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 6,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      itemID="3"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      itemID="2"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      itemID="1"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/Add.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
          }
        }
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      testID="rubric-item.description"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          View Long Description
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`RubricItem renders 1`] = `
<View
  style={
    Object {
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 6,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      itemID="3"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      itemID="2"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      itemID="1"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/Add.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
          }
        }
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      testID="rubric-item.description"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          View Long Description
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`RubricItem renders an already selected grade 1`] = `
<View
  style={
    Object {
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 6,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      itemID="3"
      on={true}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      itemID="2"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      itemID="1"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/Add.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
          }
        }
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      testID="rubric-item.description"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          View Long Description
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`RubricItem renders the selected custom grade 1`] = `
<View
  style={
    Object {
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 6,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      itemID="3"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      itemID="2"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      itemID="1"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade 12"
      on={true}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.customize-grade-2"
      value="12"
    >
      12
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      testID="rubric-item.description"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          View Long Description
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`RubricItem renders with free form 1`] = `
<View
  style={
    Object {
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 6,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/Add.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
          }
        }
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      testID="rubric-item.add-comment-2"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          Add Comment
        </Text>
      </View>
    </TouchableOpacity>
    <Text
      accessible={false}
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "alignSelf": "center",
            "color": "#8B969E",
            "fontSize": 12,
            "paddingHorizontal": 6,
          },
          Object {
            "fontFamily": ".SFUIDisplay",
          },
        ]
      }
    >
      •
    </Text>
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      testID="rubric-item.description"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          View Long Description
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`RubricItem will not show the add comment button and will show the comment if there is one 1`] = `
<View
  style={
    Object {
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "600",
        },
        Object {
          "fontFamily": ".SFUIDisplay-semibold",
        },
      ]
    }
  >
    A description
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 6,
        "marginTop": 12,
      }
    }
  >
    <CircleToggle
      accessibilityLabel="10 — Full Credit"
      itemID="3"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-3"
      value={10}
    >
      10
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="5 — Partial Credit"
      itemID="2"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-2"
      value={5}
    >
      5
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="0 — No Credit"
      itemID="1"
      on={false}
      onLongPress={[Function]}
      onPress={[Function]}
      onPressOut={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.points-1"
      value={0}
    >
      0
    </CircleToggle>
    <CircleToggle
      accessibilityLabel="Customize Grade"
      on={false}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 4,
          "marginRight": 4,
        }
      }
      testID="rubric-item.customize-grade-2"
      value=""
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/Add.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
          }
        }
      />
    </CircleToggle>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      testID="rubric-item.description"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-medium",
              },
            ]
          }
        >
          View Long Description
        </Text>
      </View>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": 8,
      }
    }
  >
    <View
      style={
        Object {
          "maxWidth": 300,
        }
      }
    >
      <Image
        capInsets={
          Object {
            "bottom": 16,
            "left": 18,
            "right": 18,
            "top": 24,
          }
        }
        resizeMode="stretch"
        source={
          Object {
            "uri": "chatBubble",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "height": null,
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": -4,
              "width": null,
            },
            Object {
              "tintColor": "#f5f5f5",
              "transform": Array [
                Object {
                  "scaleX": -1,
                },
              ],
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "paddingBottom": 8,
                "paddingHorizontal": 12,
                "paddingTop": 12,
              },
              Object {
                "color": "#2D3B45",
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        A comment
      </Text>
    </View>
    <TouchableOpacity
      accessibilityTraits="button"
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onPress={[Function]}
      style={
        Object {
          "marginLeft": 16,
        }
      }
      testID="rubric-item.edit-comment-2"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#374A59",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "color": "#73818C",
                  "fontSize": 16,
                  "fontWeight": "600",
                },
              ],
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Edit
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
